src/main/java/frc/robot/subsystems/drive/ModuleIOTalonFX.java

```diff
@@ -0,0 +1,161 @@
+// modify it under the terms of the GNU General Public License
+// version 3 as published by the Free Software Foundation or
+// available in the root directory of this project.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+package frc.robot.subsystems.drive;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.MotorOutputConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+
+/**
+ * Module IO implementation for Talon FX drive motor controller, Talon FX turn
+ * motor controller, and CANcoder
+ *
+ * <p>
+ * NOTE: This implementation should be used as a starting point and adapted to
+ * different hardware configurations (e.g. If using an analog encoder, copy from
+ * "ModuleIOSparkMax")
+ *
+ * <p>
+ * To calibrate the absolute encoder offsets, point the modules straight (such
+ * that forward motion on the drive motor will propel the robot forward) and
+ * copy the reported values from the absolute encoders using AdvantageScope.
+ * These values are logged under "/Drive/ModuleX/TurnAbsolutePositionRad"
+ */
+public class ModuleIOTalonFX implements ModuleIO
+{
+	private final TalonFX driveTalon;
+	private final TalonFX turnTalon;
+	private final CANcoder cancoder;
+
+	private final StatusSignal<Double> drivePosition;
+	private final StatusSignal<Double> driveVelocity;
+	private final StatusSignal<Double> driveAppliedVolts;
+	private final StatusSignal<Double> driveCurrent;
+
+	private final StatusSignal<Double> turnAbsolutePosition;
+	private final StatusSignal<Double> turnPosition;
+	private final StatusSignal<Double> turnVelocity;
+	private final StatusSignal<Double> turnAppliedVolts;
+	private final StatusSignal<Double> turnCurrent;
+
+	// Gear ratios for SDS MK4i L2, adjust as necessary
+
+	private final boolean isTurnMotorInverted = true;
+	private final Rotation2d absoluteEncoderOffset;
+	private final double DRIVE_GEAR_RATIO;
+	private final double TURN_GEAR_RATIO;
+
+	public ModuleIOTalonFX(int driveId, int turnId, int canId, double turnGearRatio, double driveGearRatio)
+	{
+		DRIVE_GEAR_RATIO = driveGearRatio;
+		TURN_GEAR_RATIO = turnGearRatio;
+		driveTalon = new TalonFX(driveId);
+		turnTalon = new TalonFX(turnId);
+		cancoder = new CANcoder(canId);
+		absoluteEncoderOffset = new Rotation2d(0.0); // TODO: MUST BE CALIBRATED
```

Comment:
There are many ways to do this. Don't make this a code value. Either use preferences [here](https://docs.wpilib.org/en/stable/docs/software/basic-programming/robot-preferences.html#reading-preferences) or refresh a configuration, then load a configuration.